### PR TITLE
[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #44074 [quant][graphmode][fx] Support quantization for custom module
* **#43990 [quant][eagermode][refactor] Add set/get method for quantization and fusion mappings**

Summary:
Allow user to register custom quantization and fusion patterns

Custom quantization API:
- register_static_quant_module_mapping
- get_static_quant_module_mappings
- get_static_quant_module_class

- register_dynamic_quant_module_mapping
- get_dynamic_quant_module_mappings

- register_qat_module_mapping
- get_qat_module_mappings

for fx graph mode quantization
- register_quantized_operator
- get_quantized_operator

Custom fusion API:
- register_fuser_method
- get_fuser_method

Example:

```
register_static_quant_module_mapping(torch.nn.Conv2d, torch.nn.quantized.Conv2d)
register_dynamic_quant_module_mapping(torch.nn.Conv2d, torch.nn.quantized.dynamic.Conv2d)
register_qat_module_mapping(torch.nn.Conv2d, torch.nn.qat.Conv2d)
register_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d), fuse_conv_bn)

>> get_static_quant_module_class(torch.nn.Conv2d)
torch.nn.quantized.Conv2d

>> get_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d))
fuser_conv_bn
```

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23485344](https://our.internmc.facebook.com/intern/diff/D23485344)

### CODE REVIEW

**Comment by vkuzo** [comment]
nice! any chance we could add an example usage of this API to register a custom pattern to the PR summary?  would make it easier to review

File: torch/quantization/_numeric_suite.py [hunk 1]
```diff
@@ -437,7 +422,7 @@ def compare_model_outputs(
     q_model,
     *data,
     Logger=OutputLogger,
-    white_list=DEFAULT_NUMERIC_SUITE_COMPARE_MODEL_OUTPUT_WHITE_LIST,
+    white_list=None
```
Comment (z-a-f) [inline torch/quantization/_numeric_suite.py:425]: Should we change the `white_list` while we are at it?

**Review by z-a-f — APPROVED** [review]
LGTM

File: torch/quantization/_numeric_suite.py (hunk 1 above)
Comment (jerryzh168) [reply torch/quantization/_numeric_suite.py:425]: i'm afraid this is bc breaking, would be better to change all at once

**Comment by jerryzh168** [comment]
> nice! any chance we could add an example usage of this API to register a custom pattern to the PR summary? would make it easier to review

done

File: torch/quantization/quantization_mappings.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+# mapping from floating point function or torch ops to quantized ops
+FLOAT_TO_QUANTIZED_OPERATOR_MAPPING = {
+    F.elu: torch._ops.ops.quantized.elu,
+    F.hardswish: torch._ops.ops.quantized.hardswish,
+    F.instance_norm: torch._ops.ops.quantized.instance_norm,
+    F.layer_norm: torch._ops.ops.quantized.layer_norm,
+}
+
+def register_static_quant_module_class(module_class, static_quant_module_class):
```
Comment (vkuzo) [inline torch/quantization/quantization_mappings.py:88]: optional, how about something like this to improve readability

```
def register_static_quant_module_swap(fp_source_module_class, static_quant_target_module_class):
    """Registers a module swap of {fp_source_module_class} to {static_quant_target_module_class} 
    during static quant convert step.  static_quant_target_module_class must have from_float 
    defined as a class method."""
```

Comment (jerryzh168) [reply torch/quantization/quantization_mappings.py:89]: sure, will change to `register_static_quant_module_mapping`

**Pull request closed without merging**